File: bin/wikicounter.py
import time

class WordObject:
    def __init__(self, word, count):
        self.Word = word
        self.Count = count

def sortFunc(a):
    return(a.Count)

Words = { }

f = open("words.txt")
for line in f.readlines():
    line = line[:-1]
    if(line in Words):
        Words[line] = Words[line] + 1
    else:
        Words[line] = 1
f.close()

x = list(Words.items())

array = [ ]

for i in range(Words.__len__()):
    array.append(WordObject(x[i][0], x[i][1]))

pTime = time.process_time_ns()

print("[Python] Parsed file in " + str(pTime / 1000000) + " milliseconds")

array.sort(reverse=True, key=sortFunc)

print("[Python] Sorted list in " + str((time.process_time_ns() - pTime) / 1000000) + " milliseconds")

f = open("Python_Out.txt", "w")
for i in range(array.__len__()):
    f.write(array[i].Word + ": " + str(array[i].Count) + "\n")

print("[Python] Finished in " + str((time.process_time_ns()) / 1000000) + " milliseconds")

File: bin/wikicounter.js
import {readFileSync, promises, appendFileSync as fsPromises, unlinkSync, appendFileSync} from 'fs';

var start = performance.now();

class WordObject {
    constructor(word, count) {
        this.Word = word;
        this.Count = count
    }
}

var Words = { };

const text = readFileSync("words.txt", 'utf-8');
const lines = text.split("\n");

lines.forEach(element => {
    if(element in Words) {
        Words[element]++;
    } else {
        Words[element] = 1
    }
});

var w = Object.keys(Words);
var c = Object.values(Words);

var array = [];

for(let i = 0; i < w.length; i++) {
    array[i] = new WordObject(w[i], c[i]);
}

var pEnd = performance.now()

console.log("[Javascript] Parsed file in " + (pEnd - start) + " milliseconds");

array.sort(function(a, b) { return parseInt(b.Count) - parseInt(a.Count); });

console.log("[Javascript] Sorted list in " + (performance.now() - pEnd) + " milliseconds");

for(let i = 0; i < array.length; i++) {
    var buf = "" + array[i].Word + ": " + array[i].Count + "\n";
    appendFileSync("Javascript_Out.txt", buf);
}

console.log("[Javascript] Finished in " + (performance.now() - start) + " milliseconds");